implementation/clabbert/src/main/java/pwr/jakprzyjade/clabbert/api/middlewares/UserHeaderInterceptor.java

```diff
@@ -0,0 +1,73 @@
+/* @jakubzehner (C) 2024 */
+package pwr.jakprzyjade.clabbert.api.middlewares;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import pwr.jakprzyjade.clabbert.api.annotations.UserRoles;
+import pwr.jakprzyjade.clabbert.domain.UserRole;
+import pwr.jakprzyjade.clabbert.domain.exceptions.authorization.UserIdHeaderMissing;
+import pwr.jakprzyjade.clabbert.domain.exceptions.authorization.UserRoleHeaderMissing;
+import pwr.jakprzyjade.clabbert.domain.exceptions.authorization.UserRoleNotSupported;
+import pwr.jakprzyjade.clabbert.domain.exceptions.authorization.UserUnauthorized;
+
+@Component
+public class UserHeaderInterceptor implements HandlerInterceptor {
+
+    private static final List<String> supportedRoles =
+            Arrays.stream(UserRole.values()).map(Enum::name).toList();
+
+    @Override
+    public boolean preHandle(
+            @NonNull HttpServletRequest request,
+            @NonNull HttpServletResponse response,
+            @NonNull Object handler)
+            throws Exception {
+        var userId = request.getHeader("user-id");
+        var userRole = request.getHeader("user-role");
```

Comment:
These are my proposed header names. What do you think of them?
@tchojnacki @mlodybercik @piterek130 

I also want to point out that, contrary to what @mlodybercik  said, requiring `X-` as the beginning of the header name is [deprecated](https://stackoverflow.com/a/3561399).

Reply:
Okay, it would hard to find a better source than RFC, so I guess the `X-` is out of the picture now. Good find 👍

Reply:
If we omit the `X-`, how about a different prefix to differentiate between our headers and standard ones? Like `jp-user-id`/`jp-user-role` (**J**ak**P**rzyjade). It's weird that `user-role` is controlled by use while `user-agent` isn't.

Reply:
We also need more conventions regarding, for instance is guest expressed as `user-role: null` or `"guest"`? What is the value of `user-id` for guests? `null`? empty? or no header at all? I'm fine with documenting it in Markdown myself, but there are still many undecided factors.

Reply:
I am then changing it to:
jp-user-id
jp-user-role
As discussed.